File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/constraints/Passable.java
package com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.constraints;

import com.anathema_roguelike.environment.Location;

public class Passable implements TargetConstraint<Location, Character> {

	@Override
	public boolean apply(Location target, Character character) {
		return target.isPassable();
	}
}

File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/constraints/Passable.scala
package com.anathema_roguelike
package entities.characters.perks.actions.targetingstrategies.constraints

import com.anathema_roguelike.environment.Location
import com.anathema_roguelike.entities.characters.Character

class Passable extends TargetConstraint[Location, Character] {
  override def apply(target: Location, character: Character): Boolean = target.isPassable
}